src/Symfony/Component/Console/Event/ConsoleSignalEvent.php

```diff
@@ -21,15 +21,36 @@
 final class ConsoleSignalEvent extends ConsoleEvent
 {
     private int $handlingSignal;
+    private int|false $exitCode;
 
-    public function __construct(Command $command, InputInterface $input, OutputInterface $output, int $handlingSignal)
+    public function __construct(Command $command, InputInterface $input, OutputInterface $output, int $handlingSignal, int|false $exitCode = 0)
     {
         parent::__construct($command, $input, $output);
         $this->handlingSignal = $handlingSignal;
+        $this->exitCode = $exitCode;
     }
 
     public function getHandlingSignal(): int
     {
         return $this->handlingSignal;
     }
+
+    public function setExitCode(int $exitCode): void
+    {
+        if ($exitCode < 0 || $exitCode > 255) {
+            throw new \InvalidArgumentException('Exit code must be between 0 and 255 or null.');
```

Comment:
This message seems weird to me, as the parameter `$exitCode` is not allowing `null` here.

Reply:
Thanks, fixed